### ISSUE TITLE
Iterating in templates

### ISSUE THREAD

**Issue by vaso123** [roundcube/roundcubemail#6960]
I am wondering there is no option to iterate through an array in templates.

I do not want to PHP in my template files, I'd like to keep the HTML format and set everything through config and environment variables.

Now I need to show tha latest news from an RSS feed. I am geting the articles, push them into an array, and on the `news.html` I'd like to show them, the title, author, date, lead, and url.

I do not found anything by google.

How is it possible with roundcube?

**Comment by alecpl** [comment]
You have to create the HTML chunk in PHP and pass it into the template via template variable or object.

https://github.com/roundcube/roundcubemail/wiki/Skin-Markup

Use `set_env()` or `add_handler()` method of `$RCMAIL->output`.

**Comment by vaso123** [comment]
Ok, thank you, I do it as you recommended. I built the HTML string, and added it to the environment variabless:

`$this->plugin->framework()->output->set_env('news', $html);`

In my `news.html` template I've try to show it:

` <roundcube:var name="env:news" />`

For some reason, it seems the output handler is converts my html entities to html encoded.

I've got this in my template:

`&lt;div class=&quot;item&quot;&gt;` and so on.

How can I avoid this?
